ci/terraform/shared/dynamodb.tf

```diff
@@ -542,6 +542,12 @@ resource "aws_dynamodb_resource_policy" "client_registry_table_policy" {
   policy       = data.aws_iam_policy_document.cross_account_table_resource_policy_document.json
 }
 
+resource "aws_dynamodb_resource_policy" "identity_credentials_table_policy" {
+  count        = var.identity_credentials_cross_account_access_enabled ? 1 : 0
+  resource_arn = aws_dynamodb_table.identity_credentials_table.arn
+  policy       = data.aws_iam_policy_document.cross_account_table_resource_policy_document.json
```

Comment:
this policy doc doesn't allow deletion, and one of the policies in the file enough is a deletion policy so a lambda will probably need that

Reply:
If adding `dynamodb:DeleteItem` to the cross-account policy, then will apply to all tables. As we won't give all lambdas permission to delete, this should be okay but shall have a look into it next week to see if it is possible to just apply it to the identity credentials

Reply:
Yeah agreed that maybe in this case we want a different policy